### PR TITLE
admin --check-database also checks database vs nominatim version

### PR DESCRIPTION
**Pull request by mtmail**
Should the test run before other tests or later? I think the hints should give instruction for lower database version or lower nominatim version.

```
nominatim admin --check-database
2023-11-02 15:12:55: Using project directory: /home/vagrant/nominatim-project
2023-11-02 15:12:55: Checking database
Checking database connection ... OK
Checking database_version matches Nominatim software version ... Failed
Database version (4.3.0-0) doesn't match Nominatim version (4.1.0-0)

Hints:
* Are you connecting to the right database?
* Have you run all migrations?

Project directory: /home/vagrant/nominatim-project
Current setting of NOMINATIM_DATABASE_DSN: pgsql:dbname=nominatim
```

### CODE REVIEW

**Comment by lonvia** [comment]
That's the right place to run it.

The hints should definitely distinguish between db_version < nominatim_version ("Please migrate") and vice versa ("You must not run an older Nominatim on a newer database. Upgrade your software.")

File: test/python/tools/test_check_database.py [hunk 1]
```diff
@@ -10,25 +10,36 @@
...
 
 from nominatim.tools import check_database as chkdb
+import nominatim.version
 
 def test_check_database_unknown_db(def_config, monkeypatch):
     monkeypatch.setenv('NOMINATIM_DATABASE_DSN', 'pgsql:dbname=fjgkhughwgh2423gsags')
     assert chkdb.check_database(def_config) == 1
 
 
-def test_check_database_fatal_test(def_config, temp_db):
```
Comment (lonvia) [inline test/python/tools/test_check_database.py:19]: This was intentional. the 'temp_db' creates an empty database.

File: nominatim/tools/check_database.py [hunk 2]
```diff
@@ -146,11 +148,46 @@ def check_connection(conn: Any, config: Configuration) -> CheckResult:
...
+@_check(hint="""\
+             Database version ({db_version}) doesn't match Nominatim version ({nom_version})
+
+             Hints:
+             * Are you connecting to the correct database?
+             
+             A) If database version is lower than Nominatim version: Run migrations
+                nominatim admin --migrate
+ 
+             B) If Nominatim version is lower than database version: Update Nominatim
```
Comment (lonvia) [inline nominatim/tools/check_database.py:160]: Hand in the appropriate hint in a variable, then you don't have to confuse people with the two possibilities.

Comment (mtmail) [reply nominatim/tools/check_database.py]: addednow

File: test/python/tools/test_check_database.py (hunk 1 above)
Comment (mtmail) [reply test/python/tools/test_check_database.py]: ok, removed again

**Comment by lonvia** [comment]
Can you please rebase on master? That will solve the CI fails.

File: nominatim/tools/check_database.py [hunk 3]
```diff
@@ -146,11 +148,52 @@ def check_connection(conn: Any, config: Configuration) -> CheckResult:
...
+        if db_version == NOMINATIM_VERSION:
+            return CheckState.OK
+
+        instruction = (
+            'Run migrations: nominatim admin --migrate'
+            if db_version < NOMINATIM_VERSION
+            else 'You need to upgrade the Nominatim software.'
+        )
+    else:
+        instruction = None
```
Comment (lonvia) [inline nominatim/tools/check_database.py:185]: You want an empty string here or it will say 'None' in the error message.

**Comment by mtmail** [comment]
@lonvia Rebased now and indeed tests are passing.

**Pull request merged**